### PR TITLE
[ModuleInterface] Print full type if ambiguous for extensions.

### PR DESCRIPTION
**Pull request by varungandhi-apple**
Fixes rdar://79093752.

Not sure why `Options.FullyQualifiedTypesIfAmbiguous` was set to `false` in the first place (I would think that setting should always be true apart from debugging?).

### CODE REVIEW

**Comment by varungandhi-apple** [comment]
@swift-ci smoke test

**Comment by varungandhi-apple** [comment]
Okay, this breaks IDE tests because more things start getting qualified... trying to figure out a different solution...

**Comment by slavapestov** [comment]
I believe there's a way to set PrintOptions specifically for printing module interfaces which doesn't affect the IDE tests. I agree that the extended type should always be fully qualified in module interfaces, just like we fully qualify TypeReprs in other contexts as well.

**Comment by varungandhi-apple** [comment]
One of the things need to make this work is to remove the forced `= false;` in `printExtendedTypeName` and one more function (so that any changes in `printSwiftInterfaceFile` would propagate correctly). However, the problem with doing that is that the IDE setting for `FullyQualifiedTypesIfAmbiguous` + weak ambiguity detection means that the IDE output still ended up changing (presumably, that's why the `= false` was present?). So I've added a new setting specifically for extended types to preserve the IDE output.

If we added better ambiguity detection in the future, then we could remove this extra setting and still have nice IDE output in the common case.

**Comment by varungandhi-apple** [comment]
@swift-ci smoke test

File: test/SourceKit/CodeComplete/complete_swiftinterface.swift [hunk 1]
```diff
@@ -19,10 +19,11 @@ let x = MyPoint(x: 1, y: 10.5)
...
 
-// CHECK: Begin completions, 5 items
+// CHECK: Begin completions, 4 items
 // CHECK: Keyword[self]/CurrNominal:          self[#MyPoint#]; name=self
 // CHECK: Decl[InstanceVar]/CurrNominal:      x[#Double#]; name=x
 // CHECK: Decl[InstanceVar]/CurrNominal:      y[#Double#]; name=y
 // CHECK: Decl[InstanceVar]/CurrNominal:      magnitudeSquared[#Double#]; name=magnitudeSquared
-// CHECK: Decl[InstanceVar]/CurrNominal:      magnitude[#Double#]; name=magnitude
+// FIXME: The following completion is missing.
+// CHECK-NOT: Decl[InstanceVar]/CurrNominal:      magnitude[#Double#]; name=magnitude
```
[the remark this answers is not in the document]
Comment (varungandhi-apple) [reply test/SourceKit/CodeComplete/complete_swiftinterface.swift]: False alarm, it's not a regression, it was the "classic" problem of module name = type name, which confused SourceKit.

**Comment by varungandhi-apple** [comment]
@swift-ci smoke test

**Comment by varungandhi-apple** [comment]
@swift-ci smoke test

https://github.com/apple/llvm-project/pull/3040

**Comment by varungandhi-apple** [comment]
@swift-ci test

https://github.com/apple/llvm-project/pull/3040

**Comment by varungandhi-apple** [comment]
@swift-ci clean test

apple/llvm-project#3040

**Comment by varungandhi-apple** [comment]
@swift-ci clean test

**Comment by varungandhi-apple** [comment]
@swift-ci test

apple/llvm-project#3040

**Comment by varungandhi-apple** [comment]
apple/llvm-project#3040

@swift-ci test

**Comment by varungandhi-apple** [comment]
apple/llvm-project#3040

@swift-ci test

**Comment by varungandhi-apple** [comment]
apple/llvm-project#3040

@swift-ci test macOS

**Comment by varungandhi-apple** [comment]
Got LGTM separately from Slava.

**Pull request merged**